### PR TITLE
Feature/kafkapub

### PR DESCRIPTION
**Pull request by wnichols**
This is a simple kafka publisher for flogo.  Supports plain, TLS, and SASL modes of operation.  Accepts a string as input and publishes a text message.  Returns the offset and partition to which the message was published.

### CODE REVIEW

**Comment by wnichols** [comment]
@torresashjian

File: activity/kafkapub/activity.go [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+	}
+	trustedCertFiles, err := ioutil.ReadDir(trustStore)
+	if err != nil || len(trustedCertFiles) == 0 {
+		return certPool, fmt.Errorf("Failed to read trusted certificates from [%s]  Must be a directory containing trusted certificates in PEM format", trustStore)
+	}
+	for _, trustCertFile := range trustedCertFiles {
+		fqfName := fmt.Sprintf("%s%c%s", trustStore, os.PathSeparator, trustCertFile.Name())
+		trustCertBytes, err := ioutil.ReadFile(fqfName)
+		if err != nil {
+			flogoLogger.Warnf("Failed to read trusted certificate [%s] ... continuing", trustCertFile.Name())
```
Comment (torresashjian) [inline activity/kafkapub/activity.go]: Do you want to call continue here? ioutil.ReadFile can return nil and you will be passing that to AppendCertsFromPEM, no real problem, just some extra computation

File: activity/kafkapub/activity.json [hunk 2]
```diff
@@ -0,0 +1,44 @@
+{
+  "name": "tibco-kafkapub",
+  "version": "0.0.1",
+  "type": "flogo:activity",
+  "ref": "github.com/wnichols/flogo-contrib/activity/kafkapub",
```
Comment (torresashjian) [inline activity/kafkapub/activity.json]: Should this point to github.com/TIBCOSoftware/flogo-contrib/activity/kafkapub instead?

File: activity/kafkapub/ui/activity.json [hunk 3]
```diff
@@ -0,0 +1,44 @@
+{
+  "name": "tibco-kafkapub",
+  "version": "0.0.1",
+  "type": "flogo:activity",
+  "ref": "github.com/wnichols/flogo-contrib/activity/kafkapub",
```
Comment (torresashjian) [inline activity/kafkapub/ui/activity.json]: Should this point to github.com/TIBCOSoftware/flogo-contrib/activity/kafkapub instead?

File: activity/kafkapub/activity.go [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+// Eval implements activity.Activity.Eval
+func (a *KafkaPubActivity) Eval(context activity.Context) (done bool, err error) {
+	err = initParms(a, context)
+	if err != nil {
+		flogoLogger.Errorf("Kafkapub parameters initialization got error: [%s]", err)
+		return false, err
+	}
+	defer func() {
+		if err := a.syncProducer.Close(); err != nil {
+			panic(err)
```
Comment (wnichols) [inline activity/kafkapub/activity.go]: Done

File: activity/kafkapub/activity.go [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+	if context.GetInput("truststore") != nil {
+		trustStore := context.GetInput("truststore")
+		if trustStore != nil && len(trustStore.(string)) > 0 {
+			trustPool, err := getCerts(trustStore.(string))
+			if err != nil {
+				return err
+			}
+			config := tls.Config{
+				RootCAs:            trustPool,
+				InsecureSkipVerify: true}
```
Comment (wnichols) [inline activity/kafkapub/activity.go]: Looks funny, but go fmt did not change it.  I moved the closing brace down a line.

File: activity/kafkapub/activity.go [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+		}
+		a.kafkaConfig.Net.SASL.Enable = true
+		a.kafkaConfig.Net.SASL.User = user
+		a.kafkaConfig.Net.SASL.Password = password
+		flogoLogger.Debugf("Kafkapub SASL parms initialized; user [%s]  password[########]", user)
+	}
+
+	syncProducer, err := sarama.NewSyncProducer(a.brokers, a.kafkaConfig)
+	if err != nil {
+		return fmt.Errorf("Kafkapub failed to create a SyncProducer.  Check any TLS or SASL parameters carefully.  Reason given: [%s]", err)
```
Comment (wnichols) [inline activity/kafkapub/activity.go]: Thanks, still learning go.

File: activity/kafkapub/activity.go (hunk 1 above)
Comment (wnichols) [inline activity/kafkapub/activity.go]: I didn't get  that from the doc.  Changed it and I'll do some testing.

File: activity/kafkapub/ui/activity.json (hunk 3 above)
Comment (wnichols) [inline activity/kafkapub/ui/activity.json]: Changed them all.  I had a script that switches them and I forgot to run it.  Is there a convenient way to test a contribution to flogo without switching all the url's?

File: activity/kafkapub/activity.go (hunk 6 above)
Comment (wnichols) [inline activity/kafkapub/activity.go]: p.s.  should I change these in the subscriber for consistency's sake?

File: activity/kafkapub/activity.go (hunk 5 above)
Comment (wnichols) [inline activity/kafkapub/activity.go]: Nope, fmt just changed it back.  The format looks funny but is correct.

**Pull request merged**